File: August 2025/07.py
class Solution:
    def minDifference(self, arr):
        def time_to_seconds(t):
            h, m, s = map(int, t.split(":"))
            return h * 3600 + m * 60 + s

        # Convert all times to seconds
        seconds = sorted(time_to_seconds(t) for t in arr)

        # Initialize min_diff with a large number
        min_diff = float('inf')

        # Compare adjacent times
        for i in range(1, len(seconds)):
            diff = seconds[i] - seconds[i - 1]
            min_diff = min(min_diff, diff)

        # Check wrap-around difference between last and first time
        wrap_diff = 86400 - seconds[-1] + seconds[0]  # 86400 seconds in a day
        min_diff = min(min_diff, wrap_diff)

        return min_diff

File: August 2025/07.cpp
class Solution {
  public:
    int minDifference(vector<string> &arr) {
        auto timeToSeconds = [](const string &t) {
            int h = stoi(t.substr(0, 2));
            int m = stoi(t.substr(3, 2));
            int s = stoi(t.substr(6, 2));
            return h * 3600 + m * 60 + s;
        };

        vector<int> seconds;
        for (const string &t : arr) {
            seconds.push_back(timeToSeconds(t));
        }

        sort(seconds.begin(), seconds.end());

        int minDiff = INT_MAX;
        for (int i = 1; i < seconds.size(); ++i) {
            minDiff = min(minDiff, seconds[i] - seconds[i - 1]);
        }

        // Wrap-around difference between last and first time
        int wrapDiff = 86400 - seconds.back() + seconds.front();
        minDiff = min(minDiff, wrapDiff);

        return minDiff;
    }
};
